### PR TITLE
CLOUDP-130482: Add test for atlasdeployments` with `keep` annotation

### PR DESCRIPTION
**Pull request by yzdann**
The test will make sure that accidentally deleting CRD won't have a side effect on `atlasdeployments` with `keep` annotation.

### All Submissions:
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: test/e2e/annotations_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			model.NewTestDataProvider(
+				"operator-ns-crd",
+				model.AProject{},
+				model.NewEmptyAtlasKeyType().UseDefaulFullAccess(),
+				[]string{"data/atlasdeployment_basic_with_keep_resource_policy.yaml"},
+				[]string{},
+				[]model.DBUser{},
+				30000,
+				[]func(*model.TestDataProvider){
+					actions.DeleteDeploymentCRD,
```
Comment (Sugar-pack) [inline test/e2e/annotations_test.go:62]: Could you please rename the function to something that indicates it's not just a deployment deletion? For example, DeleteDeploymentCRWithKeepAnnotation.

**Review by Sugar-pack — CHANGES_REQUESTED** [review]
Thank you. Great work) Just one small suggestion

File: test/e2e/annotations_test.go (hunk 1 above)
Comment (yzdann) [reply test/e2e/annotations_test.go]: Thanks for your review
Done in https://github.com/mongodb/mongodb-atlas-kubernetes/pull/612/commits/0ba3145ddee8b151d53e1add61a717209393ff6b

**Comment by yzdann** [comment]
> Thank you. Great work) Just one small suggestion Please add "deployment-annotations-ns" label to e2e label matrix.
> 
> https://github.com/mongodb/mongodb-atlas-kubernetes/blob/f6b19bde90577c4380cd75935285fc733f70edd8/.github/workflows/test-e2e.yml#L77

Done in https://github.com/mongodb/mongodb-atlas-kubernetes/commit/593d4c3ad440e2d6c917bf2df700685904a284b1

File: test/e2e/annotations_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			actions.AfterEachFinalCleanup([]model.TestDataProvider{data})
+		}
+	})
+
+	DescribeTable("Namespaced operators working only with its own namespace with different configuration",
+		func(test model.TestDataProvider) {
+			data = test
+			mainCycle(test)
+		},
+		Entry("Simple configuration with keep resource policy annotation on deployment", Label("ns-crd"),
```
Comment (Sugar-pack) [inline test/e2e/annotations_test.go:52]: Please add "deployment-annotations-ns" label

Comment (yzdann) [reply test/e2e/annotations_test.go:52]: Shouldn't we use the upper scope annotation to run all tests as a group?
https://github.com/mongodb/mongodb-atlas-kubernetes/blob/593d4c3ad440e2d6c917bf2df700685904a284b1/test/e2e/annotations_test.go#L17

Comment (Sugar-pack) [reply test/e2e/annotations_test.go:52]: Yeah. Just trying to figure out why your tests aren't launching.

**Review by Sugar-pack — APPROVED** [review]
Great work

**Review by fabritsius — APPROVED** [review]
Great Work 👍
Tests have [passed in the branch-copy](https://github.com/mongodb/mongodb-atlas-kubernetes/actions/runs/2954785890) so we can merge this as is.

**Pull request merged**